### PR TITLE
Migrate go-bitswap

### PR DESCRIPTION
**Pull request by Jorropo**
Replace #6 

Todo:
- [x] PR to update go-bitswap to stub out types: https://github.com/ipfs/go-bitswap/pull/597
- [x] Test the stubbed out types with Kubo to make sure they work: https://github.com/ipfs/kubo/pull/9450
  - Sharness and unit tests should pass, linter should fail (due to deprecation notices)
- [ ] Merge PR to stub out types in go-bitswap, and make its final release
- [ ] Once this is merged, move over go-bitswap issues and close PRs (using [the tooling](https://github.com/guseggert/repo-migration-tools)), then archive go-bitswap
- [ ] Close the Kubo PR to test stubs (no reason to merge)

### CODE REVIEW

File: internal/test/flaky.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+import (
+	"os"
+	"testing"
+)
+
+// Flaky will skip the test if the FLAKY environment variable is not empty.
+func Flaky(t *testing.T) {
+	// We can't use flags because it fails for tests that does not import this package
+	if os.Getenv("FLAKY") != "" {
```
Comment (guseggert) [inline internal/test/flaky.go]: shouldn't this be `==` ?

Comment (Jorropo) [reply internal/test/flaky.go]: I did the same mistake originally, not it must not.
This is the `return` path, so not `Skip`.
We want to run the test when `FLAKY=1`. `"1" != "" => true` so we `return`, *without* skipping.

Comment (guseggert) [reply internal/test/flaky.go]: ah yeah, maybe a better name then since we both made the same mistake

Comment (Jorropo) [reply internal/test/flaky.go]: Renamed to `RUN_FLAKY_TESTS`

**Review by guseggert — APPROVED** [review]

**Comment by Jorropo** [comment]
Migrated issues:
```console
$ python3.11 migrate.py issues   --source-repo ipfs/go-bitswap  --dest-repo ipfs/go-libipfs
Transferred issue from https://github.com/ipfs/go-bitswap/issues/599 to https://github.com/ipfs/go-libipfs/issues/66
Transferred issue from https://github.com/ipfs/go-bitswap/issues/592 to https://github.com/ipfs/go-libipfs/issues/67
Transferred issue from https://github.com/ipfs/go-bitswap/issues/591 to https://github.com/ipfs/go-libipfs/issues/68
Transferred issue from https://github.com/ipfs/go-bitswap/issues/589 to https://github.com/ipfs/go-libipfs/issues/69
Transferred issue from https://github.com/ipfs/go-bitswap/issues/587 to https://github.com/ipfs/go-libipfs/issues/70
Transferred issue from https://github.com/ipfs/go-bitswap/issues/586 to https://github.com/ipfs/go-libipfs/issues/71
Transferred issue from https://github.com/ipfs/go-bitswap/issues/576 to https://github.com/ipfs/go-libipfs/issues/72
Transferred issue from https://github.com/ipfs/go-bitswap/issues/560 to https://github.com/ipfs/go-libipfs/issues/73
Transferred issue from https://github.com/ipfs/go-bitswap/issues/559 to https://github.com/ipfs/go-libipfs/issues/74
Transferred issue from https://github.com/ipfs/go-bitswap/issues/557 to https://github.com/ipfs/go-libipfs/issues/75
Transferred issue from https://github.com/ipfs/go-bitswap/issues/556 to https://github.com/ipfs/go-libipfs/issues/76
Transferred issue from https://github.com/ipfs/go-bitswap/issues/555 to https://github.com/ipfs/go-libipfs/issues/77
Transferred issue from https://github.com/ipfs/go-bitswap/issues/554 to https://github.com/ipfs/go-libipfs/issues/78
Transferred issue from https://github.com/ipfs/go-bitswap/issues/550 to https://github.com/ipfs/go-libipfs/issues/79
Transferred issue from https://github.com/ipfs/go-bitswap/issues/548 to https://github.com/ipfs/go-libipfs/issues/80
Transferred issue from https://github.com/ipfs/go-bitswap/issues/533 to https://github.com/ipfs/go-libipfs/issues/81
Transferred issue from https://github.com/ipfs/go-bitswap/issues/532 to https://github.com/ipfs/go-libipfs/issues/82
Transferred issue from https://github.com/ipfs/go-bitswap/issues/525 to https://github.com/ipfs/go-libipfs/issues/83
Transferred issue from https://github.com/ipfs/go-bitswap/issues/510 to https://github.com/ipfs/go-libipfs/issues/84
Transferred issue from https://github.com/ipfs/go-bitswap/issues/501 to https://github.com/ipfs/go-libipfs/issues/85
Transferred issue from https://github.com/ipfs/go-bitswap/issues/493 to https://github.com/ipfs/go-libipfs/issues/86
Transferred issue from https://github.com/ipfs/go-bitswap/issues/467 to https://github.com/ipfs/go-libipfs/issues/87
Transferred issue from https://github.com/ipfs/go-bitswap/issues/443 to https://github.com/ipfs/go-libipfs/issues/88
Transferred issue from https://github.com/ipfs/go-bitswap/issues/428 to https://github.com/ipfs/go-libipfs/issues/89
Transferred issue from https://github.com/ipfs/go-bitswap/issues/419 to https://github.com/ipfs/go-libipfs/issues/90
Transferred issue from https://github.com/ipfs/go-bitswap/issues/397 to https://github.com/ipfs/go-libipfs/issues/91
Transferred issue from https://github.com/ipfs/go-bitswap/issues/391 to https://github.com/ipfs/go-libipfs/issues/92
Transferred issue from https://github.com/ipfs/go-bitswap/issues/372 to https://github.com/ipfs/go-libipfs/issues/93
Transferred issue from https://github.com/ipfs/go-bitswap/issues/370 to https://github.com/ipfs/go-libipfs/issues/94
Transferred issue from https://github.com/ipfs/go-bitswap/issues/368 to https://github.com/ipfs/go-libipfs/issues/95
Transferred issue from https://github.com/ipfs/go-bitswap/issues/109 to https://github.com/ipfs/go-libipfs/issues/96
Transferred issue from https://github.com/ipfs/go-bitswap/issues/108 to https://github.com/ipfs/go-libipfs/issues/97
Transferred issue from https://github.com/ipfs/go-bitswap/issues/81 to https://github.com/ipfs/go-libipfs/issues/98
Transferred issue from https://github.com/ipfs/go-bitswap/issues/67 to https://github.com/ipfs/go-libipfs/issues/99
Transferred issue from https://github.com/ipfs/go-bitswap/issues/55 to https://github.com/ipfs/go-libipfs/issues/100
Transferred issue from https://github.com/ipfs/go-bitswap/issues/20 to https://github.com/ipfs/go-libipfs/issues/101
Transferred issue from https://github.com/ipfs/go-bitswap/issues/1 to https://github.com/ipfs/go-libipfs/issues/102
```

**Comment by Jorropo** [comment]
```console
$ python3.11 migrate.py clean-pull-requests   --source-repo ipfs/go-bitswap  --dest-repo ipfs/go-libipfs
Closing PR: https://github.com/ipfs/go-bitswap/pull/598
Closing PR: https://github.com/ipfs/go-bitswap/pull/593
Closing PR: https://github.com/ipfs/go-bitswap/pull/590
Closing PR: https://github.com/ipfs/go-bitswap/pull/581
Closing PR: https://github.com/ipfs/go-bitswap/pull/578
Closing PR: https://github.com/ipfs/go-bitswap/pull/575
Closing PR: https://github.com/ipfs/go-bitswap/pull/573
Closing PR: https://github.com/ipfs/go-bitswap/pull/561
Closing PR: https://github.com/ipfs/go-bitswap/pull/545
Closing PR: https://github.com/ipfs/go-bitswap/pull/543
Closing PR: https://github.com/ipfs/go-bitswap/pull/529
Closing PR: https://github.com/ipfs/go-bitswap/pull/522
Closing PR: https://github.com/ipfs/go-bitswap/pull/521
Closing PR: https://github.com/ipfs/go-bitswap/pull/520
Closing PR: https://github.com/ipfs/go-bitswap/pull/519
Closing PR: https://github.com/ipfs/go-bitswap/pull/518
Closing PR: https://github.com/ipfs/go-bitswap/pull/517
Closing PR: https://github.com/ipfs/go-bitswap/pull/515
Closing PR: https://github.com/ipfs/go-bitswap/pull/512
Closing PR: https://github.com/ipfs/go-bitswap/pull/511
Closing PR: https://github.com/ipfs/go-bitswap/pull/475
Closing PR: https://github.com/ipfs/go-bitswap/pull/459
Closing PR: https://github.com/ipfs/go-bitswap/pull/414
Closing PR: https://github.com/ipfs/go-bitswap/pull/407
Closing PR: https://github.com/ipfs/go-bitswap/pull/401
Closing PR: https://github.com/ipfs/go-bitswap/pull/389
Closing PR: https://github.com/ipfs/go-bitswap/pull/360
Closing PR: https://github.com/ipfs/go-bitswap/pull/327
```

**Comment by Jorropo** [comment]
Realised it didn't migrated everything, now it crashes with 5 issues remaining:
```console
$ python3.11 migrate.py issues   --source-repo ipfs/go-bitswap  --dest-repo ipfs/go-libipfs
Transferred issue from https://github.com/ipfs/go-bitswap/issues/109 to https://github.com/ipfs/go-libipfs/issues/103
Transferred issue from https://github.com/ipfs/go-bitswap/issues/113 to https://github.com/ipfs/go-libipfs/issues/104
Transferred issue from https://github.com/ipfs/go-bitswap/issues/120 to https://github.com/ipfs/go-libipfs/issues/105
Transferred issue from https://github.com/ipfs/go-bitswap/issues/121 to https://github.com/ipfs/go-libipfs/issues/106
Transferred issue from https://github.com/ipfs/go-bitswap/issues/122 to https://github.com/ipfs/go-libipfs/issues/107
Transferred issue from https://github.com/ipfs/go-bitswap/issues/146 to https://github.com/ipfs/go-libipfs/issues/108
Transferred issue from https://github.com/ipfs/go-bitswap/issues/150 to https://github.com/ipfs/go-libipfs/issues/109
Transferred issue from https://github.com/ipfs/go-bitswap/issues/151 to https://github.com/ipfs/go-libipfs/issues/110
Transferred issue from https://github.com/ipfs/go-bitswap/issues/152 to https://github.com/ipfs/go-libipfs/issues/111
Transferred issue from https://github.com/ipfs/go-bitswap/issues/153 to https://github.com/ipfs/go-libipfs/issues/112
Transferred issue from https://github.com/ipfs/go-bitswap/issues/154 to https://github.com/ipfs/go-libipfs/issues/113
Transferred issue from https://github.com/ipfs/go-bitswap/issues/155 to https://github.com/ipfs/go-libipfs/issues/114
Transferred issue from https://github.com/ipfs/go-bitswap/issues/159 to https://github.com/ipfs/go-libipfs/issues/115
Transferred issue from https://github.com/ipfs/go-bitswap/issues/160 to https://github.com/ipfs/go-libipfs/issues/116
Transferred issue from https://github.com/ipfs/go-bitswap/issues/163 to https://github.com/ipfs/go-libipfs/issues/117
Transferred issue from https://github.com/ipfs/go-bitswap/issues/195 to https://github.com/ipfs/go-libipfs/issues/118
Transferred issue from https://github.com/ipfs/go-bitswap/issues/207 to https://github.com/ipfs/go-libipfs/issues/119
Transferred issue from https://github.com/ipfs/go-bitswap/issues/224 to https://github.com/ipfs/go-libipfs/issues/120
Transferred issue from https://github.com/ipfs/go-bitswap/issues/228 to https://github.com/ipfs/go-libipfs/issues/121
Transferred issue from https://github.com/ipfs/go-bitswap/issues/234 to https://github.com/ipfs/go-libipfs/issues/122
Transferred issue from https://github.com/ipfs/go-bitswap/issues/249 to https://github.com/ipfs/go-libipfs/issues/123
Transferred issue from https://github.com/ipfs/go-bitswap/issues/250 to https://github.com/ipfs/go-libipfs/issues/124
Transferred issue from https://github.com/ipfs/go-bitswap/issues/258 to https://github.com/ipfs/go-libipfs/issues/125
Transferred issue from https://github.com/ipfs/go-bitswap/issues/260 to https://github.com/ipfs/go-libipfs/issues/126
Transferred issue from https://github.com/ipfs/go-bitswap/issues/280 to https://github.com/ipfs/go-libipfs/issues/127
Transferred issue from https://github.com/ipfs/go-bitswap/issues/298 to https://github.com/ipfs/go-libipfs/issues/128
Transferred issue from https://github.com/ipfs/go-bitswap/issues/314 to https://github.com/ipfs/go-libipfs/issues/129
Transferred issue from https://github.com/ipfs/go-bitswap/issues/343 to https://github.com/ipfs/go-libipfs/issues/130
Traceback (most recent call last):
  File "/home/hugo/Documents/Scripts/repo-migration-tools/migrate.py", line 225, in <module>
    migrate()
  File "/home/hugo/.local/share/virtualenvs/repo-migration-tools-_sBQoFKO/lib/python3.11/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hugo/.local/share/virtualenvs/repo-migration-tools-_sBQoFKO/lib/python3.11/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/home/hugo/.local/share/virtualenvs/repo-migration-tools-_sBQoFKO/lib/python3.11/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hugo/.local/share/virtualenvs/repo-migration-tools-_sBQoFKO/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hugo/.local/share/virtualenvs/repo-migration-tools-_sBQoFKO/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hugo/Documents/Scripts/repo-migration-tools/migrate.py", line 195, in migrate_issues_cmd
    new_issue_number = transfer_issue(token, issue_id, repo_id)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hugo/Documents/Scripts/repo-migration-tools/migrate.py", line 44, in transfer_issue
    return requests.post(
           ^^^^^^^^^^^^^^
KeyError: 'data'
```

**Pull request merged**